src/managers/form/buttons/TriggerButton.js

```diff
@@ -1,30 +1,160 @@
     }
 
-    createElement (options) {
-        let className = 'wk-auth-form-button';
-        if (options) {
-            if (options.className) {
-                className += ` ${options.className}`;
+    checkOptions(options) {
+        if(!Array.isArray(options.authProviders)) return options;
+        for (const authProvider of options.authProviders) {
+            if (isEmpty(authProvider.provider)) continue;
+
+            const aProvider = authProvider.provider.toLowerCase();
+            options[aProvider] = {
+                fullLabel: '',
+                signInLabel: '',
+                signUpLabel: '',
+                label: '',
+                backgroundColor: '',
+                textColor: '',
+                textColorStyle: '',
+                iconUrl: '',
+                styles: {}
+            };
+            const { fullLabel, signInLabel, signUpLabel, buttonColor, buttonTextColor, iconUrl } = authProvider;
+            if (aProvider === 'email') {
+                options[aProvider].fullLabel = this.#fullLabel[aProvider];
+                options[aProvider].label = this.#fullLabel[aProvider];
+                options[aProvider].iconUrl = this.#emailButonIconUrl;
+            }
+            if (!isEmpty(fullLabel)) {
+                options[aProvider].fullLabel = fullLabel;
+                options[aProvider].label = fullLabel;
+            }
+            if (!isEmpty(signInLabel)) {
+                options[aProvider].signInLabel = signInLabel;
+                if (options.defaultFormSlug === SIGN_IN_FORM_SLUG) {
+                    options[aProvider].label = signInLabel;
+                }
+            }
+            if (!isEmpty(signUpLabel)) {
+                options[aProvider].signUpLabel = signUpLabel;
+                if (options.defaultFormSlug === SIGN_UP_FORM_SLUG) {
+                    options[aProvider].label = signUpLabel;
+                }
+            }
+            if (!isEmpty(buttonColor)) {
+                options[aProvider].backgroundColor = buttonColor;
+                options[aProvider].styles.backgroundColor = buttonColor;
+            }
+            if (!isEmpty(buttonTextColor )) {
+                options[aProvider].textColor = buttonTextColor;
+                options[aProvider].textColorStyle = ` style="color:${buttonTextColor}"`;
+            }
+            if (!isEmpty(iconUrl)) {
+                options[aProvider].iconUrl = iconUrl;
             }
         }
+        return options;
+    }
+
+    subscribeEventForChanges() {
+        /** change Firebase button titles on load */
+        this.events.subscribe(FIREBASE_UI_SHOWN, () => {
+            this.#changeAuthButtonsTitle(this.#options.defaultFormSlug, false);
+            this.#changeAuthButtonsTitleColor();
+        }, {once: true});
+
+        /** change titles on auth form change */
+        this.events.subscribe(DEFAULT_AUTH_FORM_SLUG_UPDATED, (value) => {
+            if (value?.new && value.new !== value.old) {
```

Comment:
do we need checking value?.new ?

Reply:
Done: Added lodash.get and rewrote part of the code using it